apps/devhub/tests/test_views.py

```diff
@@ -1124,9 +1125,32 @@ def get_step(self):
         return SubmitStep.objects.get(addon=self.get_addon())
 
 
-class TestSubmitStep1(TestSubmitBase):
+class TestAgreement(TestSubmitBase):
+    def test_agreement_first(self):
+        self.user.update(read_dev_agreement=None)
+        response = self.client.get(reverse('devhub.api_key_agreement'))
+        eq_(response.status_code, 200)
+        doc = pq(response.content)
+        eq_(doc('#breadcrumbs a').eq(1).attr('href'), reverse('devhub.addons'))
+        links = doc('#agreement-container a')
+        assert links
+        for ln in links:
+            href = ln.attrib['href']
+            assert not href.startswith('%'), (
+                "Looks like link %r to %r is still a placeholder" %
+                (href, ln.text))
+
+    def test_agreement_second(self):
+        self.user.update(read_dev_agreement=None)
+
+        response = self.client.post(reverse('devhub.api_key_agreement'),
+                                    follow=True)
+        assert response.status_code == 200
```

Comment:
This should also explicitly assert the correct redirect like `self.assertRedirects(response, reverse('devhub.api_key'))`